src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,13 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
 
-    /**
-     * Return the list of machines.
-     * In the implementation of this method please use your MachineProducer implementations
-     * See that 'Class type' is not parametrized.
-     * Consider to parametrize this Class< PARAMETRIZE_ME > with specific wildcard bounding
-     *
-     * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
-     * @return the list of machines
-     */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends Machine> type);
 
-    /**
-     * Fill the machines list with passed value
-     * Replace the Object with parametrized value.
-     * This method should be able to work well with any type of machine passed as 'value'
-     *
-     * @param machines - list of machines to be filled with value
-     * @param value    - any object of machine sub class
-     */
-    void fill(List<Object> machines, Object value);
+    void fill(List<? super Machine> machines, T value);
```

Comment:
Якщо ми параметризували інтерфейс як `<T extends Machine>` то тепер у методах ми повинні відштовхуватись від меншої межі, тобто `T` а не від `Machine`. Інакше може виникнути ситуація, коли ми інтерфейс параметризуємо наприклад треком, але при цьому в метод зможемо передати бульдозера. А по факту, якщо ми параметризовуємо треком, то метод повинен працювати тільки з треками і його наслідниками
це стосується всіх методів в інтерфейсі
```suggestion
    void fill(List<? super T> machines, T value);
```